TestBackwardsCompatibilityV1 e2e test is flakey

**Description**

The `TestBackwardsCompatibilityV1` e2e test is flakey and regularly fails on CI when trying to contact the CI's port.

**Steps to reproduce the issue:**

Run `make E2E_TESTS=TestBackwardsCompatibilityV1 -f docker.Makefile test-e2e`. Occasionally this will fail with an error saying wget failed to contact localhost:8080 or localhost:8081

**Describe the results you received:**

The test occasionally fails with the following error message. Note that it should retry the wget in a poll and only fail after 30s but instead fails immediately.

```
--- FAIL: TestBackwardsCompatibilityV1 (66.93s)
    helper_test.go:65: assertion failed: 
        Command:  docker exec -t 3ec92eaa50778efbefef894a68984031b114dc7a644ff194806658f23d706781 /usr/bin/wget -O - http://localhost:8081
        ExitCode: 1
        Error:    exit status 1
        Stdout:   Connecting to localhost:8081 (127.0.0.1:8081)
        wget: can't connect to remote host (127.0.0.1): Connection refused
        
        Stderr:   
        
        Failures:
        ExitCode was 1 expected 0
        Expected no error
```

**Describe the results you expected:**

The test should retry the wget in a poll and only fail after 30s if the port has still failed to update.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:29:52 2019
 OS/Arch:           linux/amd64
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:29:19 2019
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker app version`:**

```
Version:               v0.9.0-zeta1-220-gc51fa999db-dirty
Git commit:            c51fa999
Built:                 Thu Dec  5 13:41:25 2019
OS/Arch:               linux/amd64
```

**Output of `docker info`:**

```
Client:
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.0-zeta1-218-gd198096555)
  buildx: Build with BuildKit (Docker Inc., v0.3.1-tp-docker)

Server:
 Containers: 2
  Running: 2
  Paused: 0
  Stopped: 0
 Images: 86
 Server Version: 19.03.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: active
  NodeID: psbhhk91pf8a9jz691g56r3l5
  Is Manager: true
  ClusterID: pfic6c0yy15np9y3yb0h9zhaa
  Managers: 1
  Nodes: 1
  Default Address Pool: 10.0.0.0/8  
  SubnetSize: 24
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 0
  Autolock Managers: false
  Root Rotation In Progress: false
  Node Address: 192.168.65.3
  Manager Addresses:
   192.168.65.3:2377
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.84-microsoft-standard
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 12.39GiB
 Name: docker-desktop
 ID: 5NX5:4OZ6:CZLY:4G4G:6OXR:6DFB:W7NM:P2WR:HHTA:DJLC:FP4F:6SDD
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 63
  Goroutines: 177
  System Time: 2019-12-12T10:15:07.7659325Z
  EventsListeners: 3
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

--- a/e2e/helper_test.go
+++ b/e2e/helper_test.go
@@ -62,7 +62,6 @@ func runWithDindSwarmAndRegistry(t *testing.T, todo func(dindSwarmAndRegistryInf
 		}
 		cmd := icmd.Command(params[0], params[1:]...)
 		result := icmd.RunCmd(cmd)
-		result.Assert(t, icmd.Success)
 		return result.Combined()
 	}
 	// Func to execute docker cli commands

